analysis/ptr/pointer.go

```diff
@@ -12,20 +12,69 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package dataflow
+package ptr
 
 import (
+	"fmt"
 	"go/types"
+	"time"
 
 	"github.com/awslabs/ar-go-tools/analysis/config"
 	"github.com/awslabs/ar-go-tools/analysis/lang"
+	"github.com/awslabs/ar-go-tools/analysis/loadprogram"
+	"github.com/awslabs/ar-go-tools/analysis/summaries"
+	"github.com/awslabs/ar-go-tools/internal/funcutil/result"
 	"github.com/awslabs/ar-go-tools/internal/pointer"
 	"golang.org/x/tools/go/ssa"
 	"golang.org/x/tools/go/ssa/ssautil"
 )
 
-// This file contains functions for running the pointer analysis on a program. The pointer analysis is implemented in
-// the x/tools/go/pointer package.
+// State extends the WholeProgramState with the pointer analysis information.
+type State struct {
+	*loadprogram.State
```

Comment:
I don't think this can modify the program state so maybe it shouldn't be a pointer

Reply:
There's a lock value (atomic.Int32) in the `loadprogram.State` so it shouldn't be passed around by value